pkg/cmd/render/render.go

```diff
@@ -180,11 +186,30 @@ func (r *renderOpts) Run() error {
 			}
 		}
 	}
-	if _, err := os.Stat(filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.key")); err == nil {
-		if _, err := os.Stat(filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.pub")); err == nil {
-			renderConfig.UserProvidedBoundSASigningKey = true
+
+	boundSAPublicPath := filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.pub")
+	boundSAPrivatePath := filepath.Join(r.generic.AssetInputDir, "bound-service-account-signing-key.key")
+	_, privStatErr := os.Stat(boundSAPrivatePath)
+	if privStatErr != nil {
```

Comment:
nit: Maybe skip a level of indentation?

```golang
if os.IsNotExist(...) {
  // create and write key
} else if err != nil {
  // return error
}

Reply:
I like the "something happened, what is it" approach maybe a little bit better.